src/Controller/Component/CheckHttpCacheComponent.php

```diff
@@ -0,0 +1,54 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.4.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Controller\Component;
+
+use Cake\Controller\Component;
+use Cake\Event\EventInterface;
+
+/**
+ * Use HTTP caching headers to see if rendering can be skipped.
+ *
+ * Checks if the response can be considered different according to the request
+ * headers, and the caching response headers. If it was not modified, then the
+ * controller and view render process is skipped. And the client will get a
+ * empty response with a "304 Not Modified" header.
+ *
+ * To use this component your controller actions must set either the `Last-Modified`
+ * or `Etag` header. Without one of these headers being set this component
+ * will have no effect.
+ */
+class CheckHttpCacheComponent extends Component
+{
+    /**
+     * Before Render hook
+     *
+     * @param \Cake\Event\EventInterface $event The Controller.beforeRender event.
+     * @return void
+     */
+    public function beforeRender(EventInterface $event): void
```

Comment:
This features requires setting `Last-Modified` header for the response through the app code isn't it?

Reply:
We need to improve the documentation for this. Even for RequestHandlerComponent the docs don't clearly state what needs to be done in app code to benfit from this feature.

Reply:
That's fair. I can write a book section for the new component.